### PR TITLE
arch/riscv: Rename SCRATCH_HARTID_OFFSET to RISCV_HARTID_OFFSET

### PR DESCRIPTION
**Pull request by xiaoxiang781216**
## Summary
and fix the typo error

## Impact
Minor

## Testing
Pass CI

### CODE REVIEW

File: arch/risc-v/src/common/riscv_percpu.h [hunk 1]
```diff
@@ -39,9 +39,9 @@
  ****************************************************************************/
 
 #ifdef __ASSEMBLY__
-#define SCRATCH_HARTID_OFFSET   (0 * INT_REG_SIZE)
+#define RISCV_HARTID_OFFSET   (0 * INT_REG_SIZE)
 #else
-#define SCRATCH_HARTID_OFFSET   offsetof(riscv_percpu_s, hartid)
+#define RISCV_HARTID_OFFSET   offsetof(riscv_percpu_s, hartid)
```
Comment (pkarashchenko) [inline arch/risc-v/src/common/riscv_percpu.h:44]: What about??
```
#define RISCV_HARTID_OFFSET (0)
#ifndef __ASSEMBLY__
_Static_assert(RISCV_HARTID_OFFSET == offsetof(riscv_percpu_s, hartid), "Error");
#ensif
```
I'm not sure if it works. Maybe static assert can be placed in C file.

File: arch/risc-v/src/common/riscv_percpu.h [hunk 2]
```diff
@@ -74,7 +74,7 @@ struct riscv_percpu_s
 void riscv_percpu_init(void);
 
 /****************************************************************************
- * Name: riscv_percpu_get_addr
+ * Name: riscv_percpu_add_hart
```
Comment (pussuw) [inline arch/risc-v/src/common/riscv_percpu.h:77]: Thanks for fixing this error

File: arch/risc-v/src/common/riscv_percpu.h (hunk 1 above)
Comment (pussuw) [reply arch/risc-v/src/common/riscv_percpu.h:44]: IMO the prefix should pre PERCPU_ like the file name. The original name before I published this was riscv_scratch.c/h but I changed it and forgot to fix the names.

Comment (xiaoxiang781216) [reply arch/risc-v/src/common/riscv_percpu.h]: Both done.

**Review by pkarashchenko — APPROVED** [review]

File: arch/risc-v/src/common/riscv_percpu.h [hunk 3]
```diff
@@ -38,10 +38,11 @@
...
  ****************************************************************************/
 
-#ifdef __ASSEMBLY__
-#define SCRATCH_HARTID_OFFSET   (0 * INT_REG_SIZE)
-#else
-#define SCRATCH_HARTID_OFFSET   offsetof(riscv_percpu_s, hartid)
+#define RISCV_PERCPU_HARTID_OFFSET   (0 * INT_REG_SIZE)
+
+#ifndef __ASSEMBLY__
+static_assert(RISCV_PERCPU_HARTID_OFFSET == offsetof(riscv_percpu_s, hartid),
```
Comment (pkarashchenko) [inline arch/risc-v/src/common/riscv_percpu.h:44]: I think we need to move static assert to C file

Comment (xiaoxiang781216) [reply arch/risc-v/src/common/riscv_percpu.h]: Done.

**Review by pkarashchenko — APPROVED** [review]

**Pull request merged**